### ISSUE TITLE
about richMeun uploadImage statusCode 411

### ISSUE THREAD

**Issue by bojunshi** [line/line-bot-sdk-nodejs#41]
it's image about log 
[https://imgur.com/a/jBeit](https://imgur.com/a/jBeit)
` { Error: Request failed with status code 411
2017-12-08T06:25:58.690170+00:00 app[web.1]:     at wrapError (/app/node_modules/@line/bot-sdk/dist/http.js:11:15)
2017-12-08T06:25:58.690171+00:00 app[web.1]:     at <anonymous>
2017-12-08T06:25:58.690172+00:00 app[web.1]:     at process._tickCallback (internal/process/next_tick.js:188:7)
2017-12-08T06:25:58.690172+00:00 app[web.1]:   statusCode: 411,
2017-12-08T06:25:58.690173+00:00 app[web.1]:   statusMessage: 'Length Required',
2017-12-08T06:25:58.690173+00:00 app[web.1]:   originalError: 
2017-12-08T06:25:58.690173+00:00 app[web.1]:    { Error: Request failed with status code 411
2017-12-08T06:25:58.690174+00:00 app[web.1]:     at createError (/app/node_modules/axios/lib/core/createError.js:16:15)
2017-12-08T06:25:58.690174+00:00 app[web.1]:     at settle (/app/node_modules/axios/lib/core/settle.js:18:12)
2017-12-08T06:25:58.690175+00:00 app[web.1]:     at IncomingMessage.handleStreamEnd (/app/node_modules/axios/lib/adapters/http.js:191:11)
2017-12-08T06:25:58.690176+00:00 app[web.1]:     at emitNone (events.js:111:20)
2017-12-08T06:25:58.690176+00:00 app[web.1]:     at IncomingMessage.emit (events.js:208:7)
2017-12-08T06:25:58.690177+00:00 app[web.1]:     at endReadableNT (_stream_readable.js:1056:12)
2017-12-08T06:25:58.690177+00:00 app[web.1]:     at _combinedTickCallback (internal/process/next_tick.js:138:11)
2017-12-08T06:25:58.690178+00:00 app[web.1]:     at process._tickCallback (internal/process/next_tick.js:180:9)
2017-12-08T06:25:58.690179+00:00 app[web.1]:      config: 
2017-12-08T06:25:58.690179+00:00 app[web.1]:       { adapter: [Function: httpAdapter],
2017-12-08T06:25:58.690183+00:00 app[web.1]:         transformRequest: [Object],
2017-12-08T06:25:58.690183+00:00 app[web.1]:         transformResponse: [Object],
2017-12-08T06:25:58.690184+00:00 app[web.1]:         timeout: 0,
2017-12-08T06:25:58.690184+00:00 app[web.1]:         xsrfCookieName: 'XSRF-TOKEN',
2017-12-08T06:25:58.690185+00:00 app[web.1]:         xsrfHeaderName: 'X-XSRF-TOKEN',
2017-12-08T06:25:58.690185+00:00 app[web.1]:         maxContentLength: -1,
2017-12-08T06:25:58.690185+00:00 app[web.1]:         validateStatus: [Function: validateStatus],
2017-12-08T06:25:58.690186+00:00 app[web.1]:         headers: [Object],
2017-12-08T06:25:58.690186+00:00 app[web.1]:         method: 'post',
2017-12-08T06:25:58.690187+00:00 app[web.1]:         url: 'https://api.line.me/v2/bot/richmenu/richmenu-1580e0ba6b867071e2d04f8437774a15/content',
2017-12-08T06:25:58.690187+00:00 app[web.1]:         data: [Object] },
2017-12-08T06:25:58.690188+00:00 app[web.1]:      request: 
2017-12-08T06:25:58.690188+00:00 app[web.1]:       ClientRequest {
2017-12-08T06:25:58.690188+00:00 app[web.1]:         domain: null,
2017-12-08T06:25:58.690189+00:00 app[web.1]:         _events: [Object],
2017-12-08T06:25:58.690189+00:00 app[web.1]:         _eventsCount: 5,
2017-12-08T06:25:58.690190+00:00 app[web.1]:         _maxListeners: undefined,
2017-12-08T06:25:58.690190+00:00 app[web.1]:         output: [],
2017-12-08T06:25:58.690190+00:00 app[web.1]:         outputEncodings: [],
2017-12-08T06:25:58.690191+00:00 app[web.1]:         outputCallbacks: [],
2017-12-08T06:25:58.690191+00:00 app[web.1]:         outputSize: 0,
2017-12-08T06:25:58.690191+00:00 app[web.1]:         writable: true,
2017-12-08T06:25:58.690192+00:00 app[web.1]:         _last: true,
2017-12-08T06:25:58.690192+00:00 app[web.1]:         upgrading: false,
2017-12-08T06:25:58.690192+00:00 app[web.1]:         chunkedEncoding: true,
2017-12-08T06:25:58.690193+00:00 app[web.1]:         shouldKeepAlive: false,
2017-12-08T06:25:58.690193+00:00 app[web.1]:         useChunkedEncodingByDefault: true,
2017-12-08T06:25:58.690193+00:00 app[web.1]:         sendDate: false,
2017-12-08T06:25:58.690193+00:00 app[web.1]:         _removedConnection: false,
2017-12-08T06:25:58.690194+00:00 app[web.1]:         _removedContLen: false,
2017-12-08T06:25:58.690194+00:00 app[web.1]:         _removedTE: false,
2017-12-08T06:25:58.690194+00:00 app[web.1]:         _contentLength: null,
2017-12-08T06:25:58.690195+00:00 app[web.1]:         _hasBody: true,
2017-12-08T06:25:58.690195+00:00 app[web.1]:         _trailer: '',
2017-12-08T06:25:58.690195+00:00 app[web.1]:         finished: true,
2017-12-08T06:25:58.690196+00:00 app[web.1]:         _headerSent: true,
2017-12-08T06:25:58.690196+00:00 app[web.1]:         socket: [Object],
2017-12-08T06:25:58.690196+00:00 app[web.1]:         connection: [Object],
2017-12-08T06:25:58.690197+00:00 app[web.1]:         _header: 'POST /v2/bot/richmenu/richmenu-1580e0ba6b867071e2d04f8437774a15/content HTTP/1.1\r\nAccept: application/json, text/plain, */*\r\nContent-Type: image/png\r\nAuthorization: Bearer pgU+ORe6VMoKzyq77792rhM7SL9InUpAnczYcaDuycpao+6NTQa18DhI26G7uc6gSrSFgQJ2sPVgsQDZBdAq/TrCSG0yvxIjnH6oVkYpYK+CitVnSBZ1Yj8yX4PDGZZ0HraClufefNbh0ZTRRedYmwdB04t89/1O/w1cDnyilFU=\r\nUser-Agent: @line/bot-sdk/5.0.1\r\nHost: api.line.me\r\nConnection: close\r\nTransfer-Encoding: chunked\r\n\r\n',
2017-12-08T06:25:58.690198+00:00 app[web.1]:         _onPendingData: [Function: noopPendingOutput],
2017-12-08T06:25:58.690198+00:00 app[web.1]:         agent: [Object],
2017-12-08T06:25:58.690198+00:00 app[web.1]:         socketPath: undefined,
2017-12-08T06:25:58.690199+00:00 app[web.1]:         timeout: undefined,
2017-12-08T06:25:58.690199+00:00 app[web.1]:         method: 'POST',
2017-12-08T06:25:58.690199+00:00 app[web.1]:         path: '/v2/bot/richmenu/richmenu-1580e0ba6b867071e2d04f8437774a15/content',
2017-12-08T06:25:58.690200+00:00 app[web.1]:         _ended: true,
2017-12-08T06:25:58.690200+00:00 app[web.1]:         res: [Object],
2017-12-08T06:25:58.690200+00:00 app[web.1]:         aborted: undefined,
2017-12-08T06:25:58.690201+00:00 app[web.1]:         timeoutCb: null,
2017-12-08T06:25:58.690201+00:00 app[web.1]:         upgradeOrConnect: false,
2017-12-08T06:25:58.690201+00:00 app[web.1]:         parser: null,
2017-12-08T06:25:58.690202+00:00 app[web.1]:         maxHeadersCount: null,
2017-12-08T06:25:58.690202+00:00 app[web.1]:         _redirectable: [Object],
2017-12-08T06:25:58.690202+00:00 app[web.1]:         [Symbol(outHeadersKey)]: [Object] },
2017-12-08T06:25:58.690203+00:00 app[web.1]:      response: 
2017-12-08T06:25:58.690203+00:00 app[web.1]:       { status: 411,
2017-12-08T06:25:58.690204+00:00 app[web.1]:         statusText: 'Length Required',
2017-12-08T06:25:58.690204+00:00 app[web.1]:         headers: [Object],
2017-12-08T06:25:58.690204+00:00 app[web.1]:         config: [Object],
2017-12-08T06:25:58.690204+00:00 app[web.1]:         request: [Object],
2017-12-08T06:25:58.690205+00:00 app[web.1]:         data: '<HTML><HEAD>\n<TITLE>Bad Request</TITLE>\n</HEAD><BODY>\n<H1>Bad Request</H1>\nYour browser sent a request that this server could not understand.<P>\nReference&#32;&#35;7&#46;eee13217&#46;1512714358&#46;1def6d09\n</BODY>\n</HTML>\n' } } }
`
it's code
it's image about code
[https://imgur.com/a/7sylD](https://imgur.com/a/7sylD)
` return createMenu().then((richMenuId) => {
        console.log('user: '+source.userId)
        console.log('richMenu: '+richMenuId)
        return client.setRichMenuImage(richMenuId,fs.createReadStream('./static/rich/controller_01.png'))
 })`

it's controller_01.png
[https://imgur.com/a/CFjEg](https://imgur.com/a/CFjEg)

I don't know why it say "Length Required"
how can I fix it
and thank you last time help

**Comment by utatti** [comment]
I will check.

**Comment by utatti** [comment]
I confirmed that the following code works with no problem. I used the png image you provided.

- https://i.imgur.com/G055uQX.png

```js
const {Client} = require('@line/bot-sdk');
const {createReadStream, createWriteStream} = require('fs');

const config = {
  channelSecret: 'your_secret',
  channelAccessToken: 'your_token',
};

const client = new Client(config);

client.getRichMenuList()
  .then((menus) =>
    // clean existing menus
    Promise.all(menus.map(menu => client.deleteRichMenu(menu.richMenuId)))
  )
  .then(() =>
    // create new
    client.createRichMenu({
      size: {
        width: 2500,
        height: 1686,
      },
      selected: false,
      name: 'nice richmenu',
      chatBarText: 'touch me',
      areas: [{
        bounds: {
          x: 0,
          y: 0,
          width: 2500,
          height: 1686,
        },
        action: {
          type: 'message',
          text: 'from rich menu',
        },
      }]
    })
  )
  .then((id) =>
    // set image
    client.setRichMenuImage(id, createReadStream('./image.png'))
      // download image
      .then(() => client.getRichMenuImage(id))
      .then((readable) =>
        readable.pipe(createWriteStream('downloaded.png')))
  );
```

I also confirmed that it downloads the same image. Can you please check the code above works in your environment? Also please check if the image is properly loaded.

**Comment by bojunshi** [comment]
ok I try now

**Comment by bojunshi** [comment]
it is my code
[code](https://codepen.io/l02162010/pen/BmgMBN?editors=1010)
[https://imgur.com/a/TkVnn](https://imgur.com/a/TkVnn)

**Comment by utatti** [comment]
There is no structure with the code above, it's just `index.js` and `image.png`. After the script is executed, there should be `downloaded.png` in the project root. What do you mean with `still can't work`?

**Comment by bojunshi** [comment]
I try and try ,but it can't work
[log](https://imgur.com/a/DTVjw)

it is code, just like your.
[code](https://imgur.com/a/E4GT9)

**Comment by utatti** [comment]
I will check the code with Heroku again.

**Comment by utatti** [comment]
Fixed in #42. Please use the latest release, 5.2.0.
